### DISCUSSION TITLE
Apache Mina SSHD vulnerability in Keycloak (CVE-2021-30129)

### DISCUSSION THREAD

**Discussion by montgomery-marcus-solute** [keycloak/keycloak#11341] [Q&A] [answered]
CVE-2021-30129 (https://nvd.nist.gov/vuln/detail/CVE-2021-30129) describes a vulnerability in Apache Mina SSHD that is fixed in Apache Mina SSHD version 2.7.0. Keycloak 17.0.1 uses Apache Mina SSHD version 2.3.0. Are there any plans to upgrade Apache Mina SSHD to at least 2.7.0?

**Comment by d-koppenhagen** [comment]
There is actually another Issue with Minia SSH described in [CVE-2022-45047](https://nvd.nist.gov/vuln/detail/CVE-2022-45047) which affects 2.7.0 too.
It should be fixed with Version 2.9.2.

**Reply by notTrinity** [reply]
Using Quay.io 20.0.1. When can we expect an update for this issue?

**Reply by edwint88** [reply]
What exactly is affected in Keycloak? and where is Apache Mina SSHD used? I found just in some tests

**Reply by notTrinity** [reply]
There are a few issues:

sshd-common | 2.7.0 | java-archive | CVE-2022-45047 | Critical | []" | [Location<RealPath=/opt/keycloak/lib/lib/main/org.apache.sshd.sshd-common-2.7.0.jar" Layer="sha256:a871be5efd9c13675dfa45adb27d6845d64022c057b3ff8b5b34872d7fc61087"

sshd-common | 2.7.0 | java-archive | GHSA-fhw8-8j55-vwgq | Critical | [2.9.2]" | [Location<RealPath=/opt/keycloak/lib/lib/main/org.wildfly.security.wildfly-elytron-1.18.3.Final.jar" Layer="sha256:a871be5efd9c13675dfa45adb27d6845d64022c057b3ff8b5b34872d7fc61087">]

io.smallrye.reactive.smallrye-mutiny-vertx-uri-template | 2.27.0 | java-archive | CVE-2022-3734 | Critical | []" | [Location<RealPath=/opt/keycloak/lib/lib/main/io.smallrye.reactive.smallrye-mutiny-vertx-uri-template-2.27.0.jar" Layer="sha256:a871be5efd9c13675dfa45adb27d6845d64022c057b3ff8b5b34872d7fc61087">]"

org.apache.james.apache-mime4j-core | 0.8.3 | java-archive | CVE-2021-40525 | Critical | []" | [Location<RealPath=/opt/keycloak/lib/lib/main/org.apache.james.apache-mime4j-core-0.8.3.jar" Layer="sha256:a871be5efd9c13675dfa45adb27d6845d64022c057b3ff8b5b34872d7fc61087">]"

**Reply by notTrinity** [reply]
There should be a comment on this. It is troubling not to have a response from Keycloak when these will be available.

**Reply by edwint88** [reply]
wildfly on quarkus?

**Reply by notTrinity** [reply]
I only know about quarkus.

**Reply by mtasan** [reply]
Have you found out any solution for that vulnerability?

**Reply by abstractj** [reply, marked as answer]
The issue was fixed a long time ago, the solution is to update to the latest release of Keycloak. See: https://github.com/keycloak/keycloak/issues?q=is%3Aissue+sort%3Aupdated-desc+Mina+SSHD+is%3Aclosed
